extensions/datadog/src/command-list-dashboards.tsx

```diff
@@ -1,27 +1,108 @@
-import { ActionPanel, List, OpenInBrowserAction } from "@raycast/api";
-import { useDashboards } from "./useDashboards";
+import { ActionPanel, List, OpenInBrowserAction, getPreferenceValues, Icon } from "@raycast/api";
 import { linkDomain } from "./util";
+import fetch from "node-fetch";
+import { useEffect, useState } from "react";
+
+export interface DashboardSummaryDefinition {
+  popularity: number;
+  tags: any[];
+  is_favorite: boolean;
+  id: string;
+  icon: null;
+  integration_id: null;
+  is_shared: boolean;
+  author: Author;
+  url: string;
+  title: string;
+  modified: Date;
+  created: Date;
+  is_read_only: boolean;
+  type: string;
+}
+
+export interface Author {
+  handle: string;
+  name: string;
+}
+
+type State = {
+  dashboardsAreLoading: boolean;
+  totalDashboards: number;
+  dashboards: DashboardSummaryDefinition[];
+};
+
+type DashboardSearchApiResponse = {
+  total: number;
+  dashboards: DashboardSummaryDefinition[];
+};
 
-// noinspection JSUnusedGlobalSymbols
 export default function CommandListDashboards() {
-  const { dashboards, dashboardsAreLoading } = useDashboards();
+  const [{ dashboardsAreLoading, dashboards, totalDashboards }, setState] = useState<State>({
+    dashboards: [],
+    totalDashboards: 0,
+    dashboardsAreLoading: true,
+  });
+  const search = (query: string) => {
+    console.log("searching for dashboards", query);
+    setState(prev => ({
+      ...prev,
+      dashboardsAreLoading: true,
+    }));
+    fetch(
+      encodeURI(
+        `https://app.datadoghq.com/api/v1/dashboard_search?with_suggested=true&query=${query}&start=0&count=50&sort=`
```

Comment:
If the library has no query method and you must make the request on your own, you also need to account for the variable server.